lib/std/io/writer.zig

```diff
@@ -46,39 +46,34 @@ pub fn Writer(
         }
 
         /// Write a native-endian integer.
-        /// TODO audit non-power-of-two int sizes
         pub fn writeIntNative(self: Self, comptime T: type, value: T) Error!void {
             var bytes: [(@typeInfo(T).Int.bits + 7) / 8]u8 = undefined;
-            mem.writeIntNative(T, &bytes, value);
+            mem.writeIntNative(std.math.ByteAlignedInt(@TypeOf(value)), &bytes, std.math.byteAlignInt(value));
```

Comment:
Hey, I have an idea how you could simplify this:
We know that `@TypeOf(value) == T` and additionally I don't think `std.math.byteAlignInt(value)` is even needed as it is coerced automatically to the first argument in this case.
```zig
mem.writeIntNative(std.math.ByteAlignedInt(T), &bytes, value);
```

Reply:
Yeah, that works. I didn't think about type coercion.

Reply:
Hey, I think you forgot to replace `@TypeOf(value)` with `T` in the commit you just made.